contracts/mock/MockERC20WithFee1.sol

```diff
@@ -0,0 +1,67 @@
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+import "../libraries/ERC20.sol";
+import "../interfaces/IERC20Extended.sol";
+
+contract MockERC20WithFee1 is ERC20, IERC20Extended {
+    using SafeMath for uint256;
+
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        uint256 _totalSupply
+    ) public ERC20(_name, _symbol) {
+        _mint(msg.sender, _totalSupply);
+    }
+
+    function _transfer(
+        address from,
+        address to,
+        uint256 value
+    ) internal override {
+        uint256 transferAmount = (value * 99) / 100;
+        uint256 burnAmount = value.sub(transferAmount);
+        _burn(from, burnAmount);
+        super._transfer(from, to, transferAmount);
+    }
+
+    function transferExactDest(address _to, uint256 _value)
+        external
+        override
+        returns (bool success)
+    {
+        // srcAmount = floor(_value * 100 / 99);
+        uint256 srcAmount = (_value * 100 + 98) / 99;
```

Comment:
using plain numbers doesn't seem so clear.
would write a few constants.
like FEE_PERCENT
or even better. 
takeFee function. that also accumulates fee or sends it somewhere.

Reply:
Ok